worlds/doom_1993/Options.py

```diff
@@ -0,0 +1,84 @@
+import typing
+
+from Options import AssembleOptions, Choice, Toggle, DeathLink, DefaultOnToggle
+
+
+class Difficulty(Choice):
+    """
+    Choose the difficulty option. Those match DOOM's difficulty options.
+    baby (I'm too young to die.) double ammos, half damage, less monsters or strength.
+    easy (Hey, not too rough.) less monsters or strength.
+    medium (Hurt me plenty.) Default.
+    hard (Ultra-Violence.) More monsters or strength.
+    nightmare (Nightmare!) Monsters attack more rapidly and respawn.
+    """
+    display_name = "Difficulty"
+    option_baby = 0
+    option_easy = 1
+    option_medium = 2
+    option_hard = 3
+    option_nightmare = 4
+    default = 2
+
+
+class RandomMonsters(Choice):
+    """
+    Choose how monsters are randomized.
+    vanilla: No randomization
+    shuffle: Monsters are shuffled within the level
+    random_balanced: Monsters are completely randomized, but balanced based on existing ratio in the level. (Small monsters vs medium vs big)
+    """
+    display_name = "Random Monsters"
+    option_vanilla = 0
+    option_shuffle = 1
+    option_random_balanced = 2
+    default = 1
+
+
+class RandomItems(Choice):
+    """
+    Choose how items are randomized.
+    vanilla: No randomization
+    shuffle: Items are shuffled within the level
+    random_balanced: Items are completely randomized, but balanced based on existing ratio in the level. (Small item vs Big)
+    """
+    display_name = "Random Items"
+    option_vanilla = 0
+    option_shuffle = 1
+    option_random_balanced = 2
+    default = 1
+
+
+class AllowDeathLogic(Toggle):
+    """Some locations require a timed puzzle that can only be tried once.
+    After which, if the player failed to get it, the location cannot be checked anymore.
+    By default, no progression items are placed here. There is a way, hovewer, to still get them:
+    Get killed in the current map. The map will reset, you can now attempt the puzzle again."""
+    display_name = "Allow Death Logic"
+
+
+class Episode1(DefaultOnToggle):
+    """Knee-Deep in the Dead"""
+    display_name = "Episode 1"
+
+
+class Episode2(DefaultOnToggle):
+    """The Shores of Hell"""
+    display_name = "Episode 2"
+
+
+class Episode3(DefaultOnToggle):
+    """Inferno"""
+    display_name = "Episode 3"
+
```

Comment:
I don't believe there's an assert anywhere stating that you need to select at least one episode, though maybe I'm missing it. You may want to put one in so that it gives a clear error if you try to generate a yaml with no episodes. But this is probably unnecessary.

Reply:
It will default to just episode 1 if none is chosen. I could comment on it in Episode1 description.